src/math/exp_fcn_hifi.c

```diff
@@ -6,21 +6,33 @@
  *
  */
 
+#include <sof/math/exp_fcn.h>
 #include <sof/common.h>
 #include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
 #include <stdint.h>
 #include <stddef.h>
+#include <errno.h>
+
+#if defined(SOFM_EXPONENTIAL_HIFI3) || defined(SOFM_EXPONENTIAL_HIFI4) || \
+	    defined(SOFM_EXPONENTIAL_HIFI5)
 
-#if defined(SOFM_EXPONENTIAL_HIFI4)
+#include <xtensa/config/core-isa.h>
```

Comment:
It makes no sense to include this file again. `SOFM_` are defined in `exp_fcn.h` which already includes `core-isa.h`.

Reply:
@singalsu @andrula-song dont we have a high level Kconfig that sets the HIFI version that can be reused ?
@ShriramShastry are we using this header only for detection of HiFi version, or do we also need it for other reasons ?

Reply:
I don't understand your answer, I think you missed my (very simple) point and that we're talking across each other. Trying to rephrase my simple point:

If any `SOFM_EXPx` is defined, then `<xtensa/config/core-isa.h>` was already included by `exp_fcn.h`. So, just don't `#include` that same file  again.

More generally speaking: testing and debugging `#include` files is extremely time-consuming and difficult. So all `#ifdef` and `#include` logic must be as simple as possible and very well commented in the source code. If some pre-processor stuff is not obvious then it must be either simplified or explained in the source.

Reply:
Yup, including `isa.h` is a mistake in `exp_fcn_hifi.c`. I will include it.